sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystConf.scala

```diff
@@ -0,0 +1,413 @@
+   * yet, return `defaultValue`. This is useful when `defaultValue` in ConfigEntry is not the
+   * desired one.
+   */
+  def getConf[T](entry: ConfigEntry[T], defaultValue: T): T = {
+    require(normalConfEntries.get(entry.key) == entry, s"$entry is not registered")
+    Option(settings.get(entry.key)).map(entry.valueConverter).getOrElse(defaultValue)
+  }
+
+  /**
+   * Return the value of Spark SQL configuration property for the given key. If the key is not set
+   * yet, return `defaultValue` in [[ConfigEntry]].
+   */
+  def getConf[T](entry: ConfigEntry[T]): T = {
+    require(normalConfEntries.get(entry.key) == entry, s"$entry is not registered")
+    entry.readFrom(reader)
+  }
+
+  /**
+   * Return the value of an optional Spark SQL configuration property for the given key. If the key
+   * is not set yet, returns None.
+   */
+  def getConf[T](entry: OptionalConfigEntry[T]): Option[T] = {
+    require(normalConfEntries.get(entry.key) == entry, s"$entry is not registered")
+    entry.readFrom(reader)
+  }
+
+  /**
+   * Return the `string` value of Spark SQL configuration property for the given key. If the key is
+   * not set yet, return `defaultValue`.
+   */
+  def getConfString(key: String, defaultValue: String): String = {
+    val entry = normalConfEntries.get(key)
+    if (entry != null && defaultValue != "<undefined>") {
+      // Only verify for built-in configurations.
+      entry.valueConverter(defaultValue)
+    }
+    Option(settings.get(key)).getOrElse(defaultValue)
+  }
+
+  /**
+   * Return all the configuration properties that have been set (i.e. not the default).
+   * This creates a new copy of the config properties in the form of a Map.
+   */
+  def getAllConfs: Map[String, String] =
+    settings.synchronized { settings.asScala.toMap }
+
+  /**
+   * Return all the built-in configuration definitions. Each definition contains key, defaultValue
+   * and doc.
+   */
+  def getAllDefinedConfs: Seq[(String, String, String)] = normalConfEntries.synchronized {
+    normalConfEntries.values.asScala.filter(_.isPublic).map { entry =>
+      (entry.key, getConfString(entry.key, entry.defaultValueString), entry.doc)
+    }.toSeq
+  }
+
+  /**
+   * Return whether a given key is set in this [[CatalystConf]].
+   */
+  def contains(key: String): Boolean = {
+    settings.containsKey(key)
+  }
+
+  private def setConfInternal(key: String, value: String): Unit = {
+    settings.put(key, value)
+  }
+
+  def unsetConf(key: String): Unit = {
+    settings.remove(key)
+  }
+
+  def unsetConf(entry: ConfigEntry[_]): Unit = {
+    settings.remove(entry.key)
+  }
+
+  def clear(): Unit = {
+    settings.clear()
+  }
+
+  override def clone(): CatalystConf = {
+    val result = new CatalystConf
+    getAllConfs.foreach {
+      case(k, v) => if (v ne null) result.setConfString(k, v)
+    }
+    result
+  }
+
+  // For test only
+  private[spark] def copy[T](entry: ConfigEntry[T], value: T): CatalystConf = {
```

Comment:
Most of the content in this file is copied from `SQLConf`, except the `lazy val` for session local timezone, and the 3 `copy` methods here.